### PR TITLE
Chatroom

### PR DESCRIPTION
**Pull request by Maqrkk**
Basic chat functionality. Create rooms (public/private/protected). Chat in rooms. Leave rooms.

### CODE REVIEW

File: src/frontend/app/src/components/chatrooms/RoomsList.vue [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+}
+
+interface DataObject {
+  state: State;
+  chats: AllChats;
+  selectedChat: Chat;
+  selectedChatName: string;
+  typedPassword: string;
+  showPassword: boolean;
+  passwordIcon: any;
```
Comment (mraasvel) [inline src/frontend/app/src/components/chatrooms/RoomsList.vue:102]: unused, linter error on latest vscode version

File: src/frontend/app/src/components/chatrooms/RoomsList.vue [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        if (chat.name === name) {
+          return chat;
+        }
+      }
+    },
+    async refreshChatList() {
+      await this.getAllChats();
+    },
+  },
+  created() {
```
Comment (mraasvel) [inline src/frontend/app/src/components/chatrooms/RoomsList.vue:181]: change to mounted and removeListener on unmounted

**Comment by K1ngmar** [comment]
- user gets disconnected when sending long message?
- no error when roomname is incorrect
- username not updated when someone changes his/her username(not sure if nessecary)

**Review by mraasvel — APPROVED** [review]
Looks good, I think it's a good begin, I made some comments but we already discussed a lot

**Review by K1ngmar — APPROVED** [review]

File: src/frontend/app/src/components/chatrooms/RoomsList.vue (hunk 1 above)
Comment (rpet91) [reply src/frontend/app/src/components/chatrooms/RoomsList.vue:102]: Nice gezien. passWordIcon removed.

File: src/frontend/app/src/components/chatrooms/RoomsList.vue (hunk 2 above)
Comment (rpet91) [reply src/frontend/app/src/components/chatrooms/RoomsList.vue:181]: Mounted en unmounted toegevoegd. Cleanere code nu.

**Pull request merged**